modules/openapi-generator/src/test/resources/3_0/petstore-with-fake-endpoints-models-for-testing-httpsig-cmpsdschemas.yaml

```diff
@@ -0,0 +1,1855 @@
+      type: integer
+      enum:
+      - 0
+      - 1
+      - 2
+      default: 0
+    OuterComposite:
+      type: object
+      properties:
+        my_number:
+          $ref: '#/components/schemas/OuterNumber'
+        my_string:
+          $ref: '#/components/schemas/OuterString'
+        my_boolean:
+          $ref: '#/components/schemas/OuterBoolean'
+    OuterNumber:
+      type: number
+    OuterString:
+      type: string
+    OuterBoolean:
+      type: boolean
+      x-codegen-body-parameter-name: boolean_post_body
+    StringBooleanMap:
+      additionalProperties:
+        type: boolean
+    FileSchemaTestClass:
+      type: object
+      properties:
+        file:
+          $ref: '#/components/schemas/File'
+        files:
+          type: array
+          items:
+            $ref: '#/components/schemas/File'
+    File:
+      type: object
+      description: Must be named `File` for test.
+      properties:
+        sourceURI:
+          description: Test capitalization
+          type: string
+    _special_model.name_:
+      properties:
+        '$special[property.name]':
+          type: integer
+          format: int64
+      xml:
+        name: '$special[model.name]'
+    HealthCheckResult:
+      type: object
+      properties:
+        NullableMessage:
+          nullable: true
+          type: string
+      description: Just a string to inform instance is up and running. Make it nullable in hope to get it as pointer in generated model.
+    NullableClass:
+      type: object
+      properties:
+        integer_prop:
+          type: integer
+          nullable: true
+        number_prop:
+          type: number
+          nullable: true
+        boolean_prop:
+          type: boolean
+          nullable: true
+        string_prop:
+          type: string
+          nullable: true
+        date_prop:
+          type: string
+          format: date
+          nullable: true
+        datetime_prop:
+          type: string
+          format: date-time
+          nullable: true
+        array_nullable_prop:
+          type: array
+          nullable: true
+          items:
+            type: object
+        array_and_items_nullable_prop:
+          type: array
+          nullable: true
+          items:
+            type: object
+            nullable: true
+        array_items_nullable:
+          type: array
+          items:
+            type: object
+            nullable: true
+        object_nullable_prop:
+          type: object
+          nullable: true
+          additionalProperties:
+            type: object
+        object_and_items_nullable_prop:
+          type: object
+          nullable: true
+          additionalProperties:
+            type: object
+            nullable: true
+        object_items_nullable:
+          type: object
+          additionalProperties:
+            type: object
+            nullable: true
+      additionalProperties:
+        type: object
+        nullable: true
+    fruit:
```

Comment:
Here and lower is the only new part in this spec
I added these schemas for composed schema tests

Reply:
@sebastien-rosset should I make a new spec here or put these additions in the source spec?

Reply:
@sebastien-rosset gentle reminder I still need an answer from you here.
- Can go-experimental handle oneOf/anyOf composed models?
- Should I make a new spec here or put these additions in the source spec?

Reply:
> @sebastien-rosset gentle reminder I still need an answer from you here.
> 
> * Can go-experimental handle oneOf/anyOf composed models?
> * Should I make a new spec here or put these additions in the source spec?

go-experimental support oneOf, but I have not tried anyOf. Let me try that and get back to you.

Reply:
I tested as follows:
1. copy the content of the petstore-with-fake-endpoints-models-for-testing-httpsig-cmpsdschemas.yaml file info petstore-with-fake-endpoints-models-for-testing-with-http-signature.yaml
1. run the ./bin/openapi3/go-experimental-petstore.sh script
1. Go to samples/openapi3/client/petstore/go-experimental/
1. Inspect the generated code, especially GmFruit which has **anyOf**
1. Compile the generated code.

The generated code seems ok, though I have not inspected thoroughly. Also, the generated code is compiling.
Hence, my recommendation is that you put your new model into the existing petstore-with-fake-endpoints-models-for-testing-with-http-signature.yaml file.

Reply:
Thanks for letting me know. I will put my new models into that existing spec.

Reply:
It is done.